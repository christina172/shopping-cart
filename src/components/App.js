import { BrowserRouter } from "react-router-dom";
import Navbar from "./Navbar";
import Main from "./Main";

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <Navbar />
        <Main />
        <footer>
          <div>
            Images and photos from <a href="https://artgamma.ru/">Gamma</a>, <a href="https://www.nevskayapalitra.ru/">Nevskaya Palitra</a>, <a href="https://eshop.koh-i-noor.eu/">Koh-I-Noor</a>
          </div>
          <div>
            Icons from <a href="https://www.svgrepo.com/">SVG repo</a>
          </div>
        </footer>
      </div>
    </BrowserRouter>
  );
}

export default App;
