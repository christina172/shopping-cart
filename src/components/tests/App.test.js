import React from "react";
import { render, screen, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";
import userEvent from "@testing-library/user-event";

describe("App component", () => {
    it("renders App component", () => {
        const { container } = render(
            <App />
        );
        expect(container).toMatchSnapshot();
    });

    test('home route does not render the sticky bar', async () => {
        render(<App />)

        expect(screen.queryByText(/total/i)).not.toBeInTheDocument()
    });

    test('shop route renders the sticky bar', async () => {
        render(<App />)
        const user = userEvent.setup()

        // verify page content for default route
        expect(screen.getByText(/beginner artists and professionals alike/i)).toBeInTheDocument()

        // verify page content for expected route after navigating
        await user.click(screen.getByText(/shop/i))
        expect(screen.getByText(/items/i)).toBeInTheDocument()
    });

    test('input field appears after clicking cart icon on a product card (click 3 cards)', async () => {
        render(<App />);
        const user = userEvent.setup()

        await user.click(screen.getByText(/shop/i))
        const carts = screen.getAllByAltText("cart");
        console.log(carts.length);
        await act(async () => {
            await user.click(carts[6]);
        });
        await act(async () => {
            await user.click(carts[7]);
        });
        await act(async () => {
            await user.click(carts[11]);
        });

        expect(screen.getAllByRole("spinbutton").length).toBe(3);
    });

    test('total quantity and total sum are calculated as expected', async () => {
        render(<App />);
        const user = userEvent.setup()

        await user.click(screen.getByText(/shop/i))
        const carts = screen.getAllByAltText("cart");
        console.log(carts.length);
        await act(async () => {
            await user.click(carts[3]);
        });
        await act(async () => {
            await user.click(carts[7]);
        });
        await act(async () => {
            await user.click(carts[12]);
        });

        const add = screen.getAllByRole("button", { name: "+" });
        await act(async () => {
            await user.click(add[0]);
        });
        await act(async () => {
            await user.click(add[0]);
        });
        await act(async () => {
            await user.click(add[1]);
        });

        expect(screen.getByText("6 Items")).toBeInTheDocument();
        expect(screen.getByText("Total: 159.4 $")).toBeInTheDocument();
    });
})