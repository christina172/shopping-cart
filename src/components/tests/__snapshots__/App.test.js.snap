// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App component 1`] = `
<div>
  <div
    class="App"
  >
    <nav>
      <div
        class="logo-and-name"
      >
        <a
          href="#/"
        >
          <img
            alt="logo"
            class="logo"
            src="artist-palette-svgrepo-com.svg"
          />
        </a>
        <a
          href="#/"
        >
          <h1>
            Artistic Touch
          </h1>
        </a>
      </div>
      <ul>
        <li>
          <a
            href="#/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#/shop"
          >
            Shop
          </a>
        </li>
      </ul>
    </nav>
    <main>
      <div
        class="home"
      >
        <h2>
          Welcome to ARTISTIC TOUCH!
        </h2>
        <h3>
          ABOUT US
        </h3>
        <p>
          Our online store was founded in 2012. Since then we have been striving to provide the best customer experience by partnering with the best brands in the field, expanding the range of products and improving customer service. We specialize in artistic paints and drawing supplies. We offer a wide range of products for beginner artists and professionals alike.
        </p>
        <h3>
          BRANDS
        </h3>
        <div
          class="brands"
        >
          <img
            alt="gamma logo"
            class="gamma"
            src="gamma.jpg"
          />
          <img
            alt="nevskaya palitra logo"
            class="palitra"
            src="nevskaya-palitra.jpg"
          />
          <img
            alt="koh-i-noor logo"
            class="koh-i-noor"
            src="koh-i-noor.jpg"
          />
        </div>
        <h3>
          CONTACT
        </h3>
        <div>
          <p>
            Telephone: +208 (45) 123-45-67
          </p>
          <p>
            Email: art_touch@mail.com
          </p>
        </div>
      </div>
    </main>
    <footer>
      <div>
        Images and photos from 
        <a
          href="https://artgamma.ru/"
        >
          Gamma
        </a>
        , 
        <a
          href="https://www.nevskayapalitra.ru/"
        >
          Nevskaya Palitra
        </a>
        , 
        <a
          href="https://eshop.koh-i-noor.eu/"
        >
          Koh-I-Noor
        </a>
      </div>
      <div>
        Icons from 
        <a
          href="https://www.svgrepo.com/"
        >
          SVG repo
        </a>
      </div>
    </footer>
  </div>
</div>
`;
